lib/puma/client.rb

```diff
@@ -342,10 +346,7 @@ def setup_body
       @body_read_start = Process.clock_gettime(Process::CLOCK_MONOTONIC, :float_millisecond)
 
       if @env[HTTP_EXPECT] == CONTINUE
-        # TODO allow a hook here to check the headers before
-        # going forward
-        @io << HTTP_11_100
-        @io.flush
+        @http_expect_100_continue_header_present = true
```

Comment:
Pretty sure we don't want to remove the current behaviour of responding quickly to `Expect: 100 Continue` and that allowing the app to define the response to be something the user opt-in to (via the Puma config probably)

Reply:
Agreed!

Reply:
(Or at least, we should keep that behavior UNLESS the user/app says they want something else).